web/js/precallView.js

```diff
@@ -0,0 +1,236 @@
+/* globals setTimeout */
+!(function (exports) {
+  'use strict';
+
+  var _templateSrc = '/templates/precall.ejs';
+  var _template;
+  var _model;
+  var testMeterInterval;
+
+  var addHandlers = function () {
+    var preCallTestResults = document.getElementById('pre-call-test-results');
+
+    preCallTestResults.addEventListener('click', function (e) {
+      var elem = e.target;
+      switch (elem.id) {
+        case 'precall-close':
+          preCallTestResults.style.display = 'none';
+          break;
+        case 'retest':
+          preCallTestResults.style.display = 'none';
+          Utils.sendEvent('roomView:retest');
+          break;
+      }
+    });
+
+    var publishSettings = document.querySelector('.publish-settings');
+
+    publishSettings.addEventListener('click', function (e) {
+      var initialVideoSwitch = document.querySelector('#initialVideoSwitch');
+      var initialAudioSwitch = document.querySelector('#initialAudioSwitch');
+      var elem = e.target;
+      elem.blur();
+      // pointer-events is not working on IE so we can receive as target a child
+      elem = HTMLElems.getAncestorByTagName(elem, 'a');
+      if (!elem) {
+        return;
+      }
+      switch (elem.id) {
+        case 'initialAudioSwitch':
+          if (!initialAudioSwitch.classList.contains('activated')) {
+            setSwitchStatus(true, true, 'Audio', 'roomView:initialAudioSwitch');
+          } else {
+            setSwitchStatus(false, true, 'Audio', 'roomView:initialAudioSwitch');
+          }
+          break;
+        case 'initialVideoSwitch':
+          if (!initialVideoSwitch.classList.contains('activated')) {
+            setSwitchStatus(true, true, 'Video', 'roomView:initialVideoSwitch');
+          } else {
+            setSwitchStatus(false, true, 'Video', 'roomView:initialVideoSwitch');
+          }
+          break;
+      }
+    });
```

Comment:
Again here just a suggestion, you could do something along these lines.
```

 publishSettings.addEventListener('click', function (e) {
  var switches = {
    initialVideoSwitch: {
      name: 'Video',
      element: document.querySelector('#initialVideoSwitch'),
      event: 'roomView:initialVideoSwitch'
    },
    initialAudioSwitch: {
      name: 'Audio',
      element:  document.querySelector('#initialAudioSwitch'),
      event: 'roomView:initialAudioSwitch'
    }
  }
  var elem = e.target;
  elem.blur();
  // pointer-events is not working on IE so we can receive as target a child
  elem = HTMLElems.getAncestorByTagName(elem, 'a');
  if (elem) {
    var switch_ = switches[elem.id];
    var activated = selectedSwitch.element.classList.contains('activated');
    setSwitchStatus(!activated, true, switch_.name, switch_.event);
  }
```

Reply:
Yes, but if there were other `a` tags in the publish-settings element, this code could result a "Cannot read property 'element' of undefined" error.

Reply:
I see, same as for my other comment, more possible outcomes than the switch options 👍

Reply:
That said, I'll still support gathering variables and ending with a single call to the function.